packages/option/src/index.ts

```diff
@@ -0,0 +1,181 @@
+import { createSignal } from "solid-js";
+
+/**
+ * a utility class representing an Optional value, which can be either Some\<T> or None.
+ *
+ * @example
+ * ```ts
+ * const option = Some(5);
+ *
+ * if (option.isSome()) {
+ *   doSomething(option.unwrap())
+ * }
+ * ```
+ */
+class Option<T> {
+  #value: T | null;
+  constructor(value: T | null) {
+    this.#value = value;
+  }
+  /**
+   * Checks whether the option is the None variant.
+   * @example
+   * ```ts
+   * const userInput = None();
+   *
+   * if (userInput.isNone()) return alert("Invalid input");
+   * ```
+   */
+  isNone() {
+    return this.#value == null;
+  }
+  /**
+   * Checks whether the option is the Some variant.
+   * @example
+   * ```ts
+   * const userSubmission = Some(5);
+   *
+   * if (userSubmission.isSome()) return alert(`The user has already submitted value of ${userInput}`);
+   * ```
+   */
+  isSome() {
+    return this.#value != null;
+  }
+  /**
+   * Unwraps the stored value, possibly null.
+   * @example
+   * ```ts
+   * const count = Some(5);
+   *
+   * // safe
+   * onMount(() => {
+   *   doSomething(count.unwrap());
+   * });
+   *
+   * // very unsafe, possibly null!!!
+   * const onClick = () => {
+   *   doSomething(count.unwrap());
+   * };
+   *
+   * ```
+   */
+  unwrap() {
```

Comment:
I kinda wish that the way to get the value would just to call it. Like a signal, not a property on it.
```ts
const count = Some(5);

// get value
doSomething(count())

// check it beforehand
// (isSome could narrow down the type of the getter to () => NonNullable<T>)
isSome(count) && doSomething(count())
```